src/mqtt.rs

```diff
@@ -42,35 +47,140 @@ impl Mqtt {
 
         Self { client }
     }
+
+    fn publish_json(&mut self, topic: &str, payload: serde_json::Value) {
+        debug!("Publishing to {topic} with payload {payload}");
+
+        let payload = serde_json::to_string(&payload).unwrap();
```

Comment:
why not use the `protobuf_json` crate to publish the original response?

Reply:
did you mean regarding line 77? The `hms_state.to_json_payload` includes the scaling and the additional efficiency attribute. wasn't sure how to do that without manually implementing the json conversion

Reply:
Makes sense.